<launch>
    <!-- Most settings are in config.yaml, which is easier to edit -->
    <rosparam command="load" file="$(arg config_file)" />

    <group ns="arm_duty_cycle">
        <node name="arm" pkg="phyto_arm" type="arm_duty_cycle.py" />

        <node name="ctd" pkg="phyto_arm" type="mock_ctd.py" />

        <node name="profiler" pkg="phyto_arm" type="mock_profiler_node.py" />

        <node name="ifcb_runner" pkg="phyto_arm" type="mock_ifcb_runner.py" />

        <!-- Map launch arg to parameter for driving arm behavior -->
        <param name="winch_enabled" value="$(arg duty_cycle_winch)" type="bool" />

        <!-- Conditionally run the motor, winch, and profiler nodes based on the winch_enabled argument -->
        <group if="$(arg duty_cycle_winch)">
            <node name="winch" pkg="phyto_arm" type="mock_winch_node.py" />
        </group>
    </group>
</launch>